packages/mui-joy/src/utils/useSlot.ts

```diff
@@ -0,0 +1,165 @@
+import * as React from 'react';
+import { ClassValue } from 'clsx';
+import { unstable_useForkRef as useForkRef } from '@mui/utils';
+import {
+  appendOwnerState,
+  resolveComponentProps,
+  unstable_mergeSlotProps as mergeSlotProps,
```

Comment:
@michaldudak I have to export the `mergeSlotProps` from MUI Base to reuse in Joy UI.

Because Joy UI components accept `component` inside slotProps, I have to create this custom hook.

Reply:
> I think this hook should be in Joy UI which is separate from MUI Base because styled-components does not exist in MUI Base.

Sure, I agree. It's better to create a utility well-suited for Joy then try to hack something with Base-specific implementation.